intranet/apps/signage/views.py

```diff
@@ -26,3 +31,95 @@ def signage_display(request, display_id):
     context["sign"] = sign
     context["page_args"] = (sign, request)
     return render(request, "signage/base.html", context)
+
+
+def eighth(request):
+    block_id = request.GET.get("block_id")
+    block_increment = 0
+    internal_ip = check_internal_ip(request)
+    if internal_ip:
+        return internal_ip
+
+    if block_id is None:
+        block = EighthBlock.objects.get_first_upcoming_block()
+        if block is None:
+            last_block = EighthBlock.objects.order_by("date").last()
+            if last_block is not None:
+                block = last_block
+    else:
+        try:
+            block = (EighthBlock.objects.prefetch_related("eighthscheduledactivity_set").get(id=block_id))
+        except EighthBlock.DoesNotExist:
+            if EighthBlock.objects.count() == 0:
+                # No blocks have been added yet
+                return render(request, "eighth/display.html", {"no_blocks": True})
+            else:
+                # The provided block_id is invalid
+                raise http.Http404
+
+    if "block_increment" in request.GET:
```

Comment:
It would be nice to have a comment explaining what `block_increment` is, since it takes a bit of time to realize what it is.